src/Command/CleanUpCommand.php

```diff
@@ -51,15 +52,19 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $query = $this->coverRepository->getIsNotUploadedQuery($limit);
         /** @var Cover $cover */
         foreach ($query->toIterable() as $cover) {
-            if ($this->coverStoreService->exists($cover->getMaterial()->getIsIdentifier())) {
-                $this->coverStoreService->removeLocalFile($cover);
+            try {
+                if ($this->coverStoreService->exists($cover->getMaterial()->getIsIdentifier())) {
+                    $this->coverStoreService->removeLocalFile($cover);
 
-                $item = $this->coverStoreService->search($cover->getMaterial()->getIsIdentifier());
-                if (null !== $item) {
-                    $cover->setRemoteUrl($item->getUrl());
+                    $item = $this->coverStoreService->search($cover->getMaterial()->getIsIdentifier());
+                    if (null !== $item) {
+                        $cover->setRemoteUrl($item->getUrl());
+                    }
+                    $cover->setUploaded(true);
+                    $this->entityManager->flush();
                 }
-                $cover->setUploaded(true);
-                $this->entityManager->flush();
+            } catch (UninitializedPropertyException $e) {
+                // Don't do anything as we have covers without material.
```

Comment:
Not obvious where the exception is throw. Also not fan of handling this with `try/catch`. Exception handling should be for actual exceptions. If a cover legitimately doesn't need to have a `Material` then this is not an exception. If we don't want to return `?Material` and do a `null` check then `Cover` should have a `hasMaterial(): bool`.

Reply:
It's an side effect of the way you first upload a cover and then in another request link it to material.